### PR TITLE
refactor(protocol-designer): Remove last instances of FieldConnector

### LINKED ISSUE #7295: PD: remove uses of deprecated HoverTooltip
The `HoverTooltip` wrapper component is deprecated, we want to use `useHoverTooltip` hook instead. Looks like we're down to around 5 instances of `HoverTooltip`. This should be a pure refactor, no user-facing differences.

### PR DESCRIPTION
**Pull request by Kadee80**
# Overview
Closes #7295 byt removing the last uses of `FieldConnector` which was the last use of `HoverTooltip`

# Changelog

- refactor(protocol-designer): Remove last instances of `FieldConnector`
- refactor(protocol-designer): Remove utils which are now unused since `FieldConnector` is gone
- up for discussion (updating documentation)

# Review requests

- [ ] Code review
- [ ] `ToggleRowField` still works as expected (see thermocycler form)
- [ ] TC commands still export correctly

# Risk assessment
PD Medium - This is the last instance of FieldConnector, but we need to make sure we didnt break any core functionality in the TC form.

### CODE REVIEW

File: protocol-designer/docs/STEP_FORMS_AND_PROTOCOL_TIMELINE.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ The rest of this doc will describe the PD-specific side of things: how Step Form
 
 **Masking values**: Masking is a behavior where a field rejects updates when they fail to meet a certain condition -- for example, a field only intended for integers will reject changes to add a decimal point. Maskers are used in the presentational layer (specifically, in the `FieldConnector` component) where they should be applied to all updates.
 
-**Field-level errors**: `steplist/fieldLevel` allows you to specify a `getErrors` function for each field. A field can have multiple "error checker" functions that can be composed together; the final result is an array of strings where each represents an error in the field. (NOTE: if there are multiple field-level errors, `FieldConnector` will just join them with `', '`.)
+**Field-level errors**: `steplist/fieldLevel` allows you to specify a `getErrors` function for each field. A field can have multiple "error checker" functions that can be composed together; the final result is an array of strings where each represents an error in the field. ~~(NOTE: if there are multiple field-level errors, `FieldConnector` will just join them with `', '`.)~~
```
Comment (Kadee80) [inline protocol-designer/docs/STEP_FORMS_AND_PROTOCOL_TIMELINE.md:71]: Wasn't sure how to update this properly, FieldConnector is referenced

Comment (IanLondon) [reply protocol-designer/docs/STEP_FORMS_AND_PROTOCOL_TIMELINE.md:71]: We have a ticket in the sprint to rewrite these docs, this is a good start to it though! Haha

File: protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js [hunk 2]
```diff
@@ -5,28 +5,25 @@ import cx from 'classnames'
...
-        disabled={disabled || props.disabled}
-        className={cx(styles.toggle_field, props.className)}
-        value={!!value}
-        onChange={(e: SyntheticInputEvent<*>) => updateValue(!value)}
-      />
-    )}
-  />
-)
+export const ToggleRowField = (props: ToggleRowProps): React.Node => {
+  const { updateValue, value, name, offLabel, onLabel, className } = props
```
Comment (IanLondon) [inline protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js]: I think you can pass `disabled` from props down to the ToggleField?

**Review by IanLondon — CHANGES_REQUESTED** [review]
`disabled` prop? Should be available (as part of `...FieldProps`)

File: protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js (hunk 2 above)
Comment (Kadee80) [reply protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js]: Just pushed a change to pass in disabled as well. Good catch!

**Review by IanLondon — APPROVED** [review]
🐰

**Review by shlokamin — APPROVED** [review]

**Pull request merged**